vms/platformvm/service.go

```diff
@@ -1233,13 +1233,10 @@ func (s *Service) AddValidator(req *http.Request, args *AddValidatorArgs, reply
 	}
 
 	s.vm.ctx.Lock.Lock()
-	defer s.vm.ctx.Lock.Unlock()
-
```

Comment:
If we return an error (other than at the last line of this function) this will never release the lock and cause a deadlock.

(Same comment applies below)